backend/integrations/oauth/oauth.go

```diff
@@ -0,0 +1,140 @@
+package oauth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/GenerateNU/sac/backend/auth"
+	"github.com/GenerateNU/sac/backend/entities/models"
+	"github.com/google/uuid"
+	"github.com/valyala/fasthttp"
+	"gorm.io/gorm"
+)
+
+type OAuthClient struct {
+	ResourceClient 	OAuthResourceClientInterface
+}
+
+type OAuthClientInterface interface {
+	Authorize(db *gorm.DB, jwt auth.JWTClientInterface, userId *uuid.UUID) string
+	Token(requestBody models.OAuthTokenRequestBody, ctx *fasthttp.RequestCtx, userId *uuid.UUID, db *gorm.DB, jwtClient *auth.JWTClientInterface) error
+}
+
+// Return the authorization URL: 
+func (client OAuthClient) Authorize(db *gorm.DB, jwt auth.JWTClientInterface, userId *uuid.UUID) (*string, error) {
```

Comment:
we should probably be using pointer receivers over value receivers so we don't have to copy the config strings around
```suggestion
func (client *OAuthClient) Authorize(db *gorm.DB, jwt auth.JWTClientInterface, userId *uuid.UUID) (*string, error) {
```